src/actix/api/snapshot_api.rs

```diff
@@ -272,6 +280,263 @@ async fn delete_collection_snapshot(
     }
 }
 
+#[get("/collections/{collection}/shards/{shard}/snapshots")]
+async fn list_shard_snapshots(
+    toc: web::Data<TableOfContent>,
+    path: web::Path<(String, ShardId)>,
+) -> impl Responder {
+    let future = async move {
+        let (collection, shard) = path.into_inner();
+        let collection = toc.get_collection(&collection).await?;
+        let snapshots = collection.list_shard_snapshots(shard).await?;
+        Ok(snapshots)
+    };
+
+    helpers::time(future).await
+}
+
+#[post("/collections/{collection}/shards/{shard}/snapshots")]
+async fn create_shard_snapshot(
+    toc: web::Data<TableOfContent>,
+    path: web::Path<(String, ShardId)>,
+    query: web::Query<SnapshottingParam>,
+) -> impl Responder {
+    let future = async move {
+        let (collection, shard) = path.into_inner();
+        let collection = toc.get_collection(&collection).await?;
+        let snapshot = collection
+            .create_shard_snapshot(shard, &toc.temp_snapshots_path().join(SNAPSHOTS_TEMP_DIR))
+            .await?;
+
+        Ok(snapshot)
+    };
+
+    helpers::time_or_accept(future, query.wait.unwrap_or(true)).await
+}
+
+// TODO: `PUT` (same as `recover_from_snapshot`) or `POST`!?
```

Comment:
```suggestion
```

Reply:
Kinda left it on purpose, to consider it later once again.

Reply:
:+1: 

Maybe we can deprecate it eventually, and remove it all together in a major release (if we ever do).